daiquiri/datalink/utils.py

```diff
@@ -0,0 +1,32 @@
+import logging
+
+from .adapter import DatalinkAdapter
+from .models import Datalink
+
+logger = logging.getLogger(__name__)
+
+
+def update_links(resource_type, resource):
+    logger.debug('update_links %s %s', resource_type, resource)
+
+    adapter = DatalinkAdapter()
+
+    identifier = adapter.get_identifier(resource_type, resource)
+    links = adapter.get_links(resource_type, resource)
```

Comment:
I think we could still try to catch is the `resource_type` was properly declared so that while configuring an instance and making a typo in the Adapter resource_type declaration we don't get an 

```python
AttributeError("get_XYZ_list() does not exists")
```
but some 
```python
ValueError("the resource type XYZ is declared in ABCAdapter but there are no corresponding methods: get_XYZ_list(), get_XYZ_identifier(), get_XYZ_links().")
```

Reply:
hmm, without any try/exept the error is:

```
AttributeError: 'DefaultDatalinkAdapter' object has no attribute 'get_schema_identifier'
```
this is pretty ok isn't it?

Reply:
ok, after reading again, I see you point you get:

```
AttributeError: 'DefaultDatalinkAdapter' object has no attribute 'get_schema1_list'
```
I will try to add a check in the constructor.

Reply:
https://github.com/django-daiquiri/daiquiri/blob/datalink/daiquiri/datalink/adapter.py#L33 and https://github.com/django-daiquiri/daiquiri/blob/datalink/daiquiri/oai/adapter.py#L43